### DISCUSSION TITLE
Direct3D12 Raytracing Sample

### DISCUSSION THREAD

**Discussion by manuelxmarquez** [amerkoleci/Vortice.Windows#128] [Show and tell]
I got a raytracing sample to work!

![image](https://user-images.githubusercontent.com/30425852/111395719-e1b54580-8693-11eb-833c-c3d6fd4a66b2.png)

Coming from Direct3D11 it took a lot of work to learn the Direct3D12 memory management part. After finding a few bugs I finally got raytracing to work.

I followed the tutorial series from NVidia which I found to be the best reference to learn from
https://developer.nvidia.com/rtx/raytracing/dxr/dx12-raytracing-tutorial-part-1
https://developer.nvidia.com/rtx/raytracing/dxr/dx12-raytracing-tutorial-part-2

I also found NSight Graphics to be a good debugging platform since the Visual Studio one doesn't seem to work yet. I was able to check out the acceleration structures and make sure all the heap descriptors were working properly.
https://developer.nvidia.com/nsight-graphics

**Comment by amerkoleci** [comment]
There are those samples as well:
https://github.com/Jorgemagic/CSharpDirectXRaytracing

**Comment by Axiverse** [comment]
Congrats! Yeah, D3D12 is a lot more complex as far as resource management. Keep at it, I've gotten some pretty eye candy out of DXR: https://www.youtube.com/watch?v=THNkGk1blMs
